src/validator.cpp

```diff
@@ -718,117 +686,111 @@ void Validator::ValidatorImpl::validateVariable(const VariablePtr &variable, con
 
 void Validator::ValidatorImpl::validateReset(const ResetPtr &reset, const ComponentPtr &component)
 {
-    std::string orderString;
+    bool noOrder = false;
+    bool noVariable = false;
+    bool noTestVariable = false;
+    bool noTestValue = false;
+    bool noResetValue = false;
+    bool varOutsideComponent = false;
+    bool testVarOutsideComponent = false;
+    std::string varParentName;
+    std::string testVarParentName;
+
+    std::string description = "Reset in component '" + component->name() + "' ";
+
     if (reset->isOrderSet()) {
-        orderString = "with order '" + convertIntToString(reset->order()) + "'";
+        description += "with order '" + convertIntToString(reset->order()) + "', ";
     } else {
-        orderString = "does not have an order set,";
+        noOrder = true;
     }
 
-    std::string variableString;
-    std::string variableContinuation;
     if (reset->variable() == nullptr) {
-        variableString = "does not reference a variable";
-        variableContinuation = ",";
-        ErrorPtr err = std::make_shared<Error>();
-        err->setDescription("Reset in component '" + component->name() + "' " + orderString + " " + variableString + ".");
-        err->setReset(reset);
-        err->setRule(SpecificationRule::RESET_VARIABLE_REFERENCE);
-        mValidator->addError(err);
+        noVariable = true;
+    } else {
+        description += "with variable '" + reset->variable()->name() + "', ";
+        auto var = reset->variable();
+        auto varParent = std::dynamic_pointer_cast<Component>(var->parent());
```

Comment:
Does this idiom change to the owningParent() function now? or is that just for models? (also below)

Reply:
Well, no.  A variable can only have a component for a parent so if it isn't that then keep calm and carry on.

Reply:
okeydokes